src/mon/MgrMonitor.cc

```diff
@@ -515,6 +515,24 @@ bool MgrMonitor::prepare_command(MonOpRequestRef op)
     if (changed && pending_map.active_gid == 0) {
       promote_standby();
     }
+  } else if (prefix == "mgr dump") {
```

Comment:
i think it should stay in the mon since it owns the MgrMap.  (and probably it's when teh mon is down that you're most interested in dumping this :)